### ISSUE TITLE
Add a check on app block for preventing importmap

### ISSUE THREAD

**Issue by bakura10** [Shopify/theme-tools#647]
Hello!

In our recent themes I am using import map. Those are awesome.

However, while this has not happened yet, I’m pretty sure some app dev will create an app block in the head that create an import map.

Because only one import map can exist per document, if an app create an import map this will break the theme completely. An app dev might not realize it if they are testing on a theme (such as Dawn) that is not using import map.

As a prevention, theme checker should emit an error if an app block contains a script whose type is importmap (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/importmap).

Thanks

**Comment by charlespwd** [comment]
> Because only one import map can exist per document. 

@yoavweiss I vaguely recall having us having a discussion about this. Did this change after all?

**Comment by bakura10** [comment]
I didn't know about multiple import maps, that's so cool! I still think we should have at least a warning in app blocks to discourage app author to do it until multiple import maps are widely supported.

**Comment by bakura10** [comment]
I actually was not even aware of the limitation regarding a module being loaded before the importmap. That's good to know! And even easier for app dev to not realize that this may break based on the theme (Dawn does not use module anywhere, while all our newer themes do). We should play it safe here!

**Comment by bakura10** [comment]
Yes, the problem are app embed in the <head>. An app embed in the head could do this:

```
<script type="importmap">
{
  SOME STUFF
}
</script>

<script type="module" src="{{ 'something.js' | asset_url }}" />
```

The module `something.js` would be able to load modules defined in the importmap in Dawn, for instance (as Dawn does not have module and does not use importmap, it would comply with the rules of importmap).

If the same app embed would be on Prestige or Impact, it would break the theme. :D

**Comment by yoavweiss** [comment]
Multiple import maps are [shipping soon](https://groups.google.com/a/chromium.org/g/blink-dev/c/4JWXZilDG6w) in Chromium and in a [future Safari](https://github.com/WebKit/WebKit/pull/37896).

But given our legacy browser support requirements, we'd probably also need a polyfill that catches the errors multiple import maps cause in (soon-to-be) legacy browsers and recovers from them (e.g. by loading them again using [system.js](https://github.com/systemjs/systemjs)).

**Comment by yoavweiss** [comment]
@guybedford pointed out to me that [es-module-shims](https://github.com/guybedford/es-module-shims) is the right polyfill for this, and that it recently [got multiple import map support](https://github.com/guybedford/es-module-shims/pull/443).

**Comment by bakura10** [comment]
I’ve been using this polyfill for nearly 2 years on our themes using importmap, never had a problem, it works really well!
